### ISSUE TITLE
linux build error SDK 2.2.0 - please fix your toolchain docker tag in workflow

### ISSUE THREAD

**Issue by qno** [surge-synthesizer/surge-rack#656]
You need to adapt the docker image vesion because your build is not working for linux with Rack SDK 2.2.0 (this is now in the default docker image version). Or fix the Linux build issue ;-)

You need to adapt this line https://github.com/surge-synthesizer/surge-rack/blob/main/.github/workflows/build-plugin.yml#L38

from `image: ghcr.io/qno/rack-plugin-toolchain-win-linux`
to `image: ghcr.io/qno/rack-plugin-toolchain-win-linux:rack-sdk-2.1.2`
for now.

**Comment by baconpaul** [comment]
Is there any way I can debug that issue? 

This is so frustrating.

**Comment by qno** [comment]
Build in Linux with the new SDK?

**Comment by baconpaul** [comment]
Yeah ubuntu 20 with the SDK should be equivalent to the docker image?

**Comment by baconpaul** [comment]
So the plugin builds clean on ubuntu 20 against the rack 2.2.0 SDK

so how do I debug this problem? it's not one you get if you actually build the code in a real machine environment.

**Comment by qno** [comment]
The cross compiler is from Ubuntu 16.04

**Comment by baconpaul** [comment]
so can i use docker to run inside the toolchain?

like is there any way on my linux box I can modify the source and run your build?

**Comment by qno** [comment]
Yes you can just use docker.

Get the image
`docker pull  ghcr.io/qno/rack-plugin-toolchain-win-linux` and run with
` docker run --rm -it  ghcr.io/qno/rack-plugin-toolchain-win-linux`.

Then checkout your repo and build it. But if you stop the container all your changes are lost.

**Comment by qno** [comment]
I can see this

```
build@1dad792c361c:~/rack-plugin-toolchain$ find / -name "*sse2.h*"
find: ‘/etc/ssl/private’: Permission denied
/home/build/rack-plugin-toolchain/Rack-SDK-lin-x64/dep/include/simde/x86/sse2.h
/home/build/rack-plugin-toolchain/Rack-SDK-mac-x64/dep/include/simde/x86/sse2.h
/home/build/rack-plugin-toolchain/Rack-SDK-win-x64/dep/include/simde/x86/sse2.h
/home/build/rack-plugin-toolchain/surge-rack/surge/libs/simde/simde/x86/sse2.h
/home/build/rack-plugin-toolchain/surge-rack/surge/libs/simde/test/x86/test-sse2.h
```
It looks like you include the file twice from different sources, one from SDK and on from your internal sources.

**Comment by baconpaul** [comment]
Thank you

I don't actually *use* simde on linux. I wonder if my make shenanigans include it anyway. 

Very close. appreciate all the help from you and @cschol here

**Comment by baconpaul** [comment]
No I'm not including those ones in surge. They are just used for the mac build.

**Comment by baconpaul** [comment]
https://github.com/VCVRack/library/issues/745

just linking these issues. Andrew also thought it was a conflict but I don't think our linux build includes that simde so I don't think it is (but i could easily be wrong - there may be a bad makefile line).

Testing in docker now.

**Comment by qno** [comment]
Also it makes no sense at all, the include guard should prevent double inclusion of same header. Is the include guard different somehow for sse2.h?

**Comment by baconpaul** [comment]
the problem is all inside one header. its not a conflict. Check out my comment on 745.

That undef happens in something with a very wierd guard.

**Comment by baconpaul** [comment]
OK I can now reporduce the problem but ..... AHHHH! your docker image doesn't include VI

**Comment by baconpaul** [comment]
OH so interesting

If I run exactly your commands it fails in the docker image

but if in the docker image I do this

```
cd surge-rack
export RACK_DIR=../Rack-SDK-lin-x64/
make
```

the same file (FX.cpp) builds without problem

**Comment by baconpaul** [comment]
https://github.com/baconpaul/RackSDK_Ordering

@qno so this is an SDK problem. If you include rack.hpp before immintrin the build fails on linux

There's a tiny repo that shows it

I think I can work around it in surge now and will resubmit to library unless we can fix the SDK

**Comment by baconpaul** [comment]
Fixed in 7900d5d9d9856f420d6435b7d1b640f3873fad63
